ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -856,7 +856,14 @@ int ompi_osc_ucx_win_attach(struct ompi_win_t *win, void *base, size_t len) {
     int insert_index = -1, contain_index;
     int ret = OMPI_SUCCESS;
 
+    ompi_osc_ucx_lock(MPI_LOCK_EXCLUSIVE,
```

Comment:
Shouldn't we take a shared lock in [`get_dynamic_win_info`](https://github.com/open-mpi/ompi/blob/main/ompi/mca/osc/ucx/osc_ucx_comm.c#L335) to make sure readers get consistent segments?

Reply:
@devreal I think the process that is calling get_dynmic_win_info should have already had the lock on target. That's why I did not want to lock it again.

Reply:
You're right. But this won't work: the process that attaches the segment can hold a shared lock on the window, so acquiring an exclusive lock will deadlock. The osc/rdma component has two different locks for that reason

Reply:
@devreal Alright, I'll make the necessary changes. Thanks for the feedback.

Reply:
@devreal can you let me know your thoughts about the second commit?

Reply:
Yeah, I think that looks good. One last thing: it's now not just the atomicity lock anymore. So should we rename it to state lock? That might avoid some confusion for future readers...

Reply:
Thanks for the feedback, I really appreciate it. Yes, I just modified it. Please take a look at the single commit.